common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockFetchingListener.java

```diff
@@ -27,7 +27,7 @@
    * automatically. If the data will be passed to another thread, the receiver should retain()
    * and release() the buffer on their own, or copy the data to a new buffer.
    */
-  void onBlockFetchSuccess(String blockId, ManagedBuffer data);
+  void onBlockFetchSuccess(String[] blockIds, ManagedBuffer data);
```

Comment:
shall we change `onBlockFetchFailure` as well?

Reply:
Current implementation is easier and when failure, Spark will report all remaining blocks (including still fetching), it is not so meaningful, upper layer does not use this information also.
Do you think we have to?